extensions/tmdb/src/components/Backdrops.tsx

```diff
@@ -0,0 +1,102 @@
+import { ActionPanel, Action, Grid, Icon, showToast, Toast } from "@raycast/api";
+import { moviedb } from "../api";
+import { MovieImagesResponse, TvImagesResponse } from "moviedb-promise";
+import { useCachedPromise } from "@raycast/utils";
+import { useState } from "react";
+
+function Backdrops({ id, type }: { id: number; type: "movie" | "tv" }) {
+  const [selectedLanguage, setSelectedLanguage] = useState<string>("all");
+
+  const fetchBackdropsAndLanguages = async () => {
+    let response: MovieImagesResponse | TvImagesResponse;
+    if (type === "movie") {
+      response = await moviedb.movieImages({ id });
+    } else {
+      response = await moviedb.tvImages({ id });
+    }
+
+    const backdropLanguages = (response.backdrops || [])
+      .map((b) => b.iso_639_1 || "no-language")
+      .filter((value, index, self) => self.indexOf(value) === index);
+
+    const langResponse = await moviedb.languages();
+    const filteredLanguages = langResponse.filter(
+      (lang) => lang.iso_639_1 && backdropLanguages.includes(lang.iso_639_1),
+    );
+
+    if (backdropLanguages.includes("no-language")) {
+      filteredLanguages.push({ iso_639_1: "no-language", english_name: "No Language" });
+    }
+
+    return {
+      backdrops: response.backdrops || [],
+      languages: filteredLanguages.sort((a, b) => (a.english_name || "").localeCompare(b.english_name || "")),
+    };
+  };
+
+  const { data: backdropData, isLoading } = useCachedPromise(fetchBackdropsAndLanguages, [], {
```

Comment:
Given the previous comment, the `args` in `useCachedPromise` should now be the arguments that you want to give to `fecthBackdropsAndLanguages`.

In this case it would be `[id, type]` instead of `[]`

That way `useCachedPromise` knows to execute the function again when one of them changes, but also to cache the results based on this pair.